File: codes/lua/inheritance/single/person.lua
local Person = {luckyNumber = 0}
Person.__index = Person

function Person.new(name)    -- The constructor
    local instance = {name = name}
    setmetatable(instance, Person)
    return instance
end

function Person:info()
    print("My name: " .. self.name);
    print("My lucky number: " .. self.luckyNumber)
end

return Person

File: codes/php/inheritance/single/person.php
<?php
class Person {

    public $luckyNumber = 0;
    private $name;

    public function __construct( $name) {
        $this->name = $name;
    }

    public function info() {
        echo "My name: $this->name\n";
        echo "My lucky number is: $this->luckyNumber\n";
    }

    private function info2() {
        echo "Person.info2\n";
    }

}

?>
